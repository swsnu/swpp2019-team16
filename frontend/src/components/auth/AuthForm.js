import React from 'react';
import styled from 'styled-components'
import Button from '../common/Button/Button';
import TextInput from '../common/TextInput';

const AuthFormBlock = styled.div``;

AuthForm.propTypes = {};

function AuthForm({ form, onChange, onSubmit }) {
  return (
    <AuthFormBlock>
        <div>
        <TextInput
            label='username'
            type="text"
            id="email-input"
            name="username"
            margin='normal'
            variant='outlined'
            onChange={onChange}
            value={form.username}
        />
        </div>        
        <div>
        <TextInput
            label='password'
            type="password"
            id="pw-input"
            name="password"
            margin='normal'
            variant='outlined'
            onChange={onChange}
            value={form.password}
        />
        </div>
        <Button id="login-button" onClick={onSubmit} children='login'/>
        <Button id="signup-button" onClick={onSubmit} children='sign up'/>
    </AuthFormBlock>
  );
}

export default AuthForm;