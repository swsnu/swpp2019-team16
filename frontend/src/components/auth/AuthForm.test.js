import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import AuthForm from './AuthForm';

describe('<AuthForm />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(
      <AuthForm
        form={{
          username: 'yata',
          password: 'yata',
        }}
        onChange={() => {}}
        onSubmit={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});