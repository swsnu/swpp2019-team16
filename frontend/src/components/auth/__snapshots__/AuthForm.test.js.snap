// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthForm /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bDWFJH"
  >
    <input
      id="email-input"
      name="username"
      type="text"
      value="yata"
    />
    <input
      id="pw-input"
      name="password"
      type="password"
      value="yata"
    />
    <button
      id="login-button"
    >
      login
    </button>
  </div>
</div>
`;
